orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/cf/CloudFoundryServerGroupCreator.java

```diff
@@ -104,6 +106,18 @@ private Artifact resolveArtifact(StageExecution stage, Object input) {
     return artifact;
   }
 
+  private Artifact resolveManifestArtifact(StageExecution stage, Object input) {
+    DeploymentManifest manifest = mapper.convertValue(input, DeploymentManifest.class);
+    Artifact artifact =
+        artifactUtils.getBoundArtifactForStage(
+            stage, manifest.getArtifactId(), manifest.getArtifact());
+    if (artifact == null) {
+      throw new IllegalArgumentException("Unable to bind the manifest artifact");
```

Comment:
Do we have any more context for why the manifest failed to bind?

Reply:
@dogonthehorizon at this point in the code, no. But the artifactUtils.getBoundArtifactForStage has plenty of different catches for problems that could arise. This is a last resort throw

Reply:
I could, however, do something a bit more specific like : "Unable to bind the manifest artifact because artifactId or Artifact is null"

Reply:
Added more detail to the error message